urbanopt_des/modelica_results.py

```diff
@@ -175,33 +377,45 @@ def resample_and_convert_to_df(
         df_energy.index.name = "datetime"
 
         # all data combined
-        data = {
-            "datetime": time,
-            "Sewer Pump Electricity": sewer_pump,
-            "GHX Pump Electricity": ghx_pump,
-            "Distribution Pump Electricity": distribution_pump,
-        } | building_data
+        data = (
+            {
+                "datetime": time,
+                "Sewer Pump Electricity": sewer_pump,
+                "GHX Pump Electricity": ghx_pump,
+                "Distribution Pump Electricity": distribution_pump,
+            }
+            | building_data
+            | chiller_data
+            | cooling_plant_pumps
+            | boiler_data
+            | heating_plant_pumps
+        )
 
         # add in the 'other variables' if they exist
         if other_vars is not None:
             for other_var in other_vars:
                 if other_var in self.modelica_data.varNames():
-                    (_, other_var_data) = self.modelica_data.values(other_var)
-                    # check the length of the data
-                    if len(other_var_data) == len(time):
-                        data[other_var] = other_var_data
-                    else:
-                        print(f'Other var "{other_var}" length does not match {len(other_var_data)} != {len(time)}')
+                    other_var_data = self.retrieve_variable_data(other_var, len(time1))
+                    data[other_var] = other_var_data
 
         df_power = pd.pandas.DataFrame(data)
```

Comment:
😱

Reply:
haha, I should fix this and do import pandas as pd.... oof..

Reply:
You already did! 😆